opensrp-reveal/src/main/java/org/smartregister/reveal/view/EditFociBoundaryActivity.java

```diff
@@ -0,0 +1,339 @@
+package org.smartregister.reveal.view;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.design.widget.Snackbar;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import com.google.gson.JsonArray;
+import com.mapbox.android.core.permissions.PermissionsManager;
+import com.mapbox.geojson.FeatureCollection;
+import com.mapbox.geojson.Geometry;
+import com.mapbox.geojson.MultiPolygon;
+import com.mapbox.geojson.Polygon;
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.location.LocationComponent;
+import com.mapbox.mapboxsdk.location.modes.RenderMode;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.maps.Style;
+import com.mapbox.mapboxsdk.plugins.annotation.Circle;
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.smartregister.domain.Location;
+import org.smartregister.reveal.R;
+import org.smartregister.reveal.application.RevealApplication;
+import org.smartregister.reveal.contract.EditFociboundaryContract;
+import org.smartregister.reveal.presenter.EditFociBoundaryPresenter;
+import org.smartregister.reveal.util.Constants;
+import org.smartregister.reveal.util.EditBoundaryState;
+import org.smartregister.reveal.util.RevealMapHelper;
+import org.smartregister.sync.helper.LocationServiceHelper;
+
+import java.util.Collections;
+
+import io.ona.kujaku.callbacks.OnLocationComponentInitializedCallback;
+import io.ona.kujaku.layers.FillBoundaryLayer;
+import io.ona.kujaku.layers.KujakuLayer;
+import io.ona.kujaku.listeners.OnDrawingCircleClickListener;
+import io.ona.kujaku.listeners.OnKujakuLayerLongClickListener;
+import io.ona.kujaku.manager.DrawingManager;
+import timber.log.Timber;
+
+import static org.smartregister.reveal.util.Utils.getLocationBuffer;
+import static org.smartregister.reveal.util.Utils.getPixelsPerDPI;
+
+/**
+ * Created by Richard Kareko on 5/13/20.
+ */
+
+public class EditFociBoundaryActivity extends BaseMapActivity implements EditFociboundaryContract.View, OnLocationComponentInitializedCallback {
+
+    private static final String TAG = EditFociBoundaryActivity.class.getName();
+
+    private DrawingManager drawingManager;
+
+    private Toolbar toolbar;
+    private Button deleteBtn ;
+    private Button savePointBtn;
+    private Button saveBoundaryBtn;
+    private Button cancelBtn;
+
+    private RevealApplication revealApplication = RevealApplication.getInstance();
+    boolean locationComponentActive = false;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5681838)